pkg/scheduler/routes/route.go

```diff
@@ -65,15 +65,12 @@ func PredicateRoute(s *scheduler.Scheduler) httprouter.Handle {
 			}
 		}
 
-		if resultBody, err := json.Marshal(extenderFilterResult); err != nil {
-			klog.ErrorS(err, "Failed to marshal extender filter result", "result", extenderFilterResult)
-			w.Header().Set("Content-Type", "application/json")
-			w.WriteHeader(http.StatusInternalServerError)
-			w.Write([]byte(err.Error()))
-		} else {
-			w.Header().Set("Content-Type", "application/json")
-			w.WriteHeader(http.StatusOK)
-			w.Write(resultBody)
+		klog.V(5).InfoS("Returning predicate response", "result", extenderFilterResult)
+
+		err := render.New(render.Options{IndentJSON: true}).JSON(w, http.StatusOK, extenderFilterResult)
+		if err != nil {
+			klog.ErrorS(err, "Failed to write JSON response")
+			return
```

Comment:
When render.New(...).JSON fails, should write error info to client. like before code:
```
                         w.Header().Set("Content-Type", "application/json")
			w.WriteHeader(http.StatusInternalServerError)
			w.Write([]byte(err.Error()))
```